virttest/qemu_vm.py

```diff
@@ -1652,7 +1652,7 @@ def sort_key(dev):
         # here is a workaround to make all windows use only 2 sockets.
         if (vcpu_sockets and vcpu_sockets > 2 and
                 params.get("os_type") == 'windows'):
-            vcpu_sockets = 2
+            vcpu_sockets = params.get_numeric("max_vcpu_sockets", 2)
```

Comment:
I agree with @PaulYuuu that the name might not be that proper if it is Windows only, so either we can remove `params.get('os_type') == 'windows'` and specify it in Windows.cfg, or we might name it `win_max_vcpu_sockets`.

Reply:
And the logic is not correct here, if `vcpu_sockets` is set to `4` and `max_vcpu_sockets` is `6`, the `vcpu_socket` will be set to `6`, which is not expected.
It should be `vcpu_sockets > max_vcpu_sockets` instead of a hard code `vcpu_sockets > 2`.